### PR TITLE
Assorted Fixes

### PR DESCRIPTION
**Pull request by Ghrehh**
Some assorted UI fixes/upgrades to the data and design section.
Ran all these changes by design.

## Description
- Input placeholder text color was the same color as the actual input text color while your cursor was hovering over an input, which could make it a little confusing whether an input had been typed in or not, so we reduced the font color of placeholder text by 1 level.

- There was some sort of weird event update cycle going on in the grid for editing relationship columns, this was causing the name of the column to be incorrectly displayed as "Column 5" or where ever it appeared numerically. Stopping the input component from firing events while disabled fixed this.
- ~~Clicking the text of the screens section opens the search~~.
- The screen search animation was a little choppy as max-height was being applied/removed instantly alongside a 300ms height transition, adding a transition to max height fixed this issue. With this and other refactors was also able to remove the JS element of this animation and make it purely CSS. 
- Extracted simple horizontal and vertical resizing into a function that returns svelte action that can be added to the resizable element and whatever handle is being used to resize it. It can also be given an optional initial value and onChange handler if this data needs to be persisted to a server, localstorage, or any arbitrary storage.
- Applied the resize actions to the left panel of the design section, though it's only really useful if you're using some heavily nested component trees.
- Added tooltips to component icons, as when a component is renamed it's not obvious what the component is.
- Moved the `...` action menu on screen nav items in the design section to match the layout of the component nav items. Initially I spent some time trying to do the opposite, making the component nav item match closer to the screen layout, but couldn't find anything that was easily implemented.

Screenshot 2023-11-23 at 17 09 41
Screenshot 2023-11-23 at 17 09 48

## Feature branch env
[Feature Branch Link](http://fb-random-fixes.fb.qa.budibase.net)

### CODE REVIEW

**Review by aptkingston — COMMENTED** [review]
Awesome work @Ghrehh! Loads of a really nice changes here, especially refactoring the panel resizing to be generic and support horizontal resizing too. Left a few comments for small things I found.

One other thing was that the screen search input seems to be a small fixed width now, rather than take up the full width of the panel:
image

File: packages/builder/src/components/common/NavItem.svelte [hunk 1]
```diff
@@ -163,9 +166,10 @@
     justify-content: flex-start;
     align-items: center;
     gap: var(--spacing-xs);
-    width: max-content;
+    width: 100%;
```
[the remark this answers is not in the document]
Comment (Ghrehh) [reply packages/builder/src/components/common/NavItem.svelte:169]: Huh I did have a massive nested list and had specifically considered this, I wonder if a temp change I was fiddling around with when demonstrating different options to Andy made it into the actual PR. Will investigate, good catch ty

**Comment by Ghrehh** [comment]
@aptkingston Thanks for the review, addressed your feedback there. 

Your point on the search field being small, you're correct and I agree it should be changed, but as far as I can tell it was already like that before this PR, and it's now a component that is consumed elsewhere, so I'd rather update it in a separate PR if that's alright?

**Comment by aptkingston** [comment]
> @aptkingston Thanks for the review, addressed your feedback there.
> 
> Your point on the search field being small, you're correct and I agree it should be changed, but as far as I can tell it was already like that before this PR, and it's now a component that is consumed elsewhere, so I'd rather update it in a separate PR if that's alright?

Of course yea - didn't realise it existed that way upstream. Something else recently must have regressed it. Happy to leave it out for now then!

**Review by aptkingston — APPROVED** [review]

**Pull request merged**